File: app/src/main/java/com/jery/tachisyrpc/MainActivity.java
package com.jery.tachisyrpc;

import androidx.appcompat.app.AppCompatActivity;

import android.annotation.SuppressLint;
import android.app.Activity;
import android.app.ActivityManager;
import android.app.NotificationManager;
import android.content.ClipData;
import android.content.ClipboardManager;
import android.content.Context;
import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.os.PowerManager;
import android.provider.Settings;
import android.service.notification.StatusBarNotification;
import android.widget.EditText;
import android.widget.Switch;
import android.widget.Toast;

import com.google.android.material.chip.Chip;

@SuppressWarnings("RedundantCast")
@SuppressLint( "StaticFieldLeak, UseSwitchCompatOrMaterialCode, BatteryLife" )
public class MainActivity extends AppCompatActivity {

    public static Chip Token, Footer;
    public static EditText Name, State, Details;
    public static Switch Switch;
    public static Activity activity;
    public static ClipboardManager clipboard;
    public static ClipData.Item clipItem;
    public static String clipText = "Testing";

    @SuppressLint("MissingInflatedId")
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        activity = this;

        Token = (Chip) findViewById(R.id.chipToken);
        Name = (EditText) findViewById(R.id.edittextName);
        State = (EditText) findViewById(R.id.edittextState);
        Details = (EditText) findViewById(R.id.edittextDetails);  // Reference Text : Akatsuki no Yona
        Switch = (Switch) findViewById(R.id.switchRPC);
        Footer = (Chip) findViewById(R.id.chipFooter);

        if (android.os.Build.VERSION.SDK_INT <= 33) {
            clipboard = (ClipboardManager) getSystemService(Context.CLIPBOARD_SERVICE);
            if (clipboard.hasPrimaryClip())
                clipItem = clipboard.getPrimaryClip().getItemAt(0);
            if (clipItem != null) {
                clipText = clipItem.getText().toString();
                Details.setText(clipText);
            }
        }

        Switch.setChecked(isServiceRunning(MyService.class));
        Switch.setOnClickListener(v -> {
            if (Switch.isChecked()) {
                Toast.makeText(MainActivity.this, clipText, Toast.LENGTH_SHORT).show();
                startService(new Intent(MainActivity.this, MyService.class));
            } else {
                Toast.makeText(MainActivity.this, "Turning off RPC", Toast.LENGTH_SHORT).show();
                stopService(new Intent(MainActivity.this, MyService.class));
            }
        });

        Footer.setOnLongClickListener(v -> {
            Intent intent = new Intent();
            String packageName = getPackageName();
            PowerManager pm = (PowerManager) getSystemService(Context.POWER_SERVICE);
            if (pm != null && !pm.isIgnoringBatteryOptimizations(packageName)) {
                intent.setAction(Settings.ACTION_REQUEST_IGNORE_BATTERY_OPTIMIZATIONS);
                intent.setData(Uri.parse("package:" + packageName));
                startActivity(intent);
            } else {
                Toast.makeText(MainActivity.this, "Battery optimization already disabled", Toast.LENGTH_SHORT).show();
            }
            return true;
        });
    }

    public boolean isNotificationPresent() {
        NotificationManager notificationManager = (NotificationManager) getSystemService(Context.NOTIFICATION_SERVICE);
        StatusBarNotification[] notificationArray = notificationManager.getActiveNotifications();
        return notificationArray != null && notificationArray.length > 0;
    }

    public boolean isServiceRunning(Class<?> serviceClass) {
        ActivityManager manager = (ActivityManager) getSystemService(Context.ACTIVITY_SERVICE);
        for (ActivityManager.RunningServiceInfo service : manager.getRunningServices(Integer.MAX_VALUE)) {
            if (serviceClass.getName().equals(service.service.getClassName())) {
                return true;
            }
        }
        return false;
    }
}

File: app/src/main/java/com/jery/tachisyrpc/MainActivity.kt
package com.jery.tachisyrpc

import android.annotation.SuppressLint
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.google.android.material.chip.Chip
import android.widget.EditText
import android.widget.Toast
import android.content.Intent
import android.os.PowerManager
import android.app.NotificationManager
import android.app.ActivityManager
import android.app.Activity
import android.content.ClipData
import android.content.ClipboardManager
import android.net.Uri
import android.provider.Settings
import android.view.View
import android.widget.Switch

@SuppressLint("StaticFieldLeak, UseSwitchCompatOrMaterialCode, BatteryLife")
class MainActivity : AppCompatActivity() {
    @SuppressLint("MissingInflatedId", "SetTextI18n")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        activity = this
        Token = findViewById<View>(R.id.chipToken) as Chip

        //        Token.setTooltipText("<--Harry-->#4627");
        Name = findViewById<View>(R.id.edittextName) as EditText
        State = findViewById<View>(R.id.edittextState) as EditText
        Details = findViewById<View>(R.id.edittextDetails) as EditText // Reference Text : Akatsuki no Yona
        Switch = findViewById<View>(R.id.switchRPC) as Switch
        Footer = findViewById<View>(R.id.chipFooter) as Chip
        Token!!.setOnClickListener { v: View? ->
            if (Token!!.text == "OTYxNTc2MjY1MzMxMTM0NDg2.GOGIQ_.yuC6PUXa89ICuBNZEdrutD6-zaWYlfI-qMmWNQ") {
                Token!!.text =
                    "OTQ1MTk2MDc5ODQ1MDgxMTIw.GYyHzW.PPUz-WxacFdGNHtIjLqjj3hk7Nqai_sCLAH3Qo" //jery99961
                Token!!.tooltipText = "jery_js#4490"
            } else {
                Token!!.text =
                    "OTYxNTc2MjY1MzMxMTM0NDg2.GOGIQ_.yuC6PUXa89ICuBNZEdrutD6-zaWYlfI-qMmWNQ" //jery2005may
                Token!!.tooltipText = "<--Harry-->#4627"
            }
        }
        clipboard = getSystemService(CLIPBOARD_SERVICE) as ClipboardManager
        if (clipboard!!.hasPrimaryClip())
            clipItem = clipboard!!.primaryClip!!.getItemAt(0)
        if (clipItem != null) {
            clipText = clipItem!!.text.toString()
            Details!!.setText(clipText)
        }

//        Switch.setChecked(isServiceRunning(MyService.class));
        if (isServiceRunning(MyService::class.java)) {
            Details!!.setText(MyService.setDetails)
            Switch!!.isChecked = true
        }
        Switch!!.setOnClickListener { v: View? ->
            if (Switch!!.isChecked) {
                Toast.makeText(this@MainActivity, Details!!.text.toString(), Toast.LENGTH_SHORT)
                    .show()
                startService(Intent(this@MainActivity, MyService::class.java))
            } else {
                Toast.makeText(this@MainActivity, "Turning off RPC", Toast.LENGTH_SHORT).show()
                stopService(Intent(this@MainActivity, MyService::class.java))
            }
        }
        Footer!!.setOnLongClickListener { v: View? ->
            val intent = Intent()
            val packageName = packageName
            val pm = getSystemService(POWER_SERVICE) as PowerManager
            if (pm != null && !pm.isIgnoringBatteryOptimizations(packageName)) {
                intent.action = Settings.ACTION_REQUEST_IGNORE_BATTERY_OPTIMIZATIONS
                intent.data = Uri.parse("package:$packageName")
                startActivity(intent)
            } else {
                Toast.makeText(
                    this@MainActivity,
                    "Battery optimization already disabled",
                    Toast.LENGTH_SHORT
                ).show()
            }
            true
        }
    }

    val isNotificationPresent: Boolean
        get() {
            val notificationManager = getSystemService(NOTIFICATION_SERVICE) as NotificationManager
            val notificationArray = notificationManager.activeNotifications
            return notificationArray != null && notificationArray.size > 0
        }

    private fun isServiceRunning(serviceClass: Class<*>): Boolean {
        val manager = getSystemService(ACTIVITY_SERVICE) as ActivityManager
        for (service in manager.getRunningServices(Int.MAX_VALUE)) {
            if (serviceClass.name == service.service.className) {
                return true
            }
        }
        return false
    }

    companion object {
        @JvmField
        var Token: Chip? = null
        var Footer: Chip? = null
        @JvmField
        var Name: EditText? = null
        @JvmField
        var State: EditText? = null
        @JvmField
        var Details: EditText? = null
        var Switch: Switch? = null
        var activity: Activity? = null
        var clipboard: ClipboardManager? = null
        var clipItem: ClipData.Item? = null
        var clipText = "Testing"
    }
}